cmd/mattermost/commands/group.go

```diff
@@ -0,0 +1,305 @@
+	RunE:    teamGroupListCmdF,
+}
+
+func init() {
+	ChannelGroupCmd.AddCommand(
+		ChannelGroupEnableCmd,
+		ChannelGroupDisableCmd,
+		ChannelGroupStatusCmd,
+		ChannelGroupListCmd,
+	)
+
+	TeamGroupCmd.AddCommand(
+		TeamGroupEnableCmd,
+		TeamGroupDisableCmd,
+		TeamGroupStatusCmd,
+		TeamGroupListCmd,
+	)
+
+	GroupCmd.AddCommand(
+		ChannelGroupCmd,
+		TeamGroupCmd,
+	)
+
+	RootCmd.AddCommand(GroupCmd)
+}
+
+func channelGroupEnableCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	channel := getChannelFromChannelArg(a, args[0])
+	if channel == nil {
+		return errors.New("Unable to find channel '" + args[0] + "'")
+	}
+
+	groups, appErr := a.GetGroupsByChannel(channel.Id, 0, 9999)
+	if appErr != nil {
+		return appErr
+	}
+
+	if len(groups) == 0 {
+		return errors.New("Channel '" + args[0] + "' has no groups associated. It cannot be group-constrained")
+	}
+
+	channel.GroupConstrained = model.NewBool(true)
+	if _, appErr = a.UpdateChannel(channel); appErr != nil {
+		return appErr
+	}
+
+	return nil
+}
+
+func channelGroupDisableCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	channel := getChannelFromChannelArg(a, args[0])
+	if channel == nil {
+		return errors.New("Unable to find channel '" + args[0] + "'")
+	}
+
+	channel.GroupConstrained = model.NewBool(false)
+	if _, appErr := a.UpdateChannel(channel); appErr != nil {
+		return appErr
+	}
+
+	return nil
+}
+
+func channelGroupStatusCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	channel := getChannelFromChannelArg(a, args[0])
+	if channel == nil {
+		return errors.New("Unable to find channel '" + args[0] + "'")
+	}
+
+	if channel.GroupConstrained != nil && *channel.GroupConstrained {
+		fmt.Println("Enabled")
+	} else {
+		fmt.Println("Disabled")
+	}
+
+	return nil
+}
+
+func channelGroupListCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	channel := getChannelFromChannelArg(a, args[0])
+	if channel == nil {
+		return errors.New("Unable to find channel '" + args[0] + "'")
+	}
+
+	groups, appErr := a.GetGroupsByChannel(channel.Id, 0, 9999)
+	if appErr != nil {
+		return appErr
+	}
+
+	for _, group := range groups {
+		fmt.Println(group.DisplayName)
+	}
+
+	return nil
+}
+
+func teamGroupEnableCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	team := getTeamFromTeamArg(a, args[0])
+	if team == nil {
+		return errors.New("Unable to find team '" + args[0] + "'")
+	}
+
+	groups, appErr := a.GetGroupsByTeam(team.Id, 0, 9999)
```

Comment:
[This PR](https://github.com/mattermost/mattermost-server/pull/10701/files#diff-0b7dcefeb566578492d6be5ea6b2453cR976) adds the ability to bypass the pagination for this query. And there's a similar one incoming for `(*App).GetGroupsByChannel`.

Reply:
I will open a follow up ticket to improve this when your PRs are merged. Thanks for the heads up!

Reply:
I'll need to fix it to merge my PRs or else the build checks wouldn't let me merge, so no need to do anything on your end.